Description: Update to support pep-0476
 This pep (included in 2.7.9) changes the behaviour of
 SSL certificate chain handling to be more py3 like.
 .
 Include required new exception behaviour in the list
 of exceptions to translate under py2.
 .
 https://github.com/python/peps/blob/master/pep-0476.txt
Author: James Page <james.page@ubuntu.com>
Forwarded: no

--- a/glanceclient/common/https.py
+++ b/glanceclient/common/https.py
@@ -149,7 +149,11 @@ class VerifiedHTTPSConnection(HTTPSConne
         if six.PY3:
             excp_lst = (TypeError, FileNotFoundError, ssl.SSLError)
         else:
-            excp_lst = ()
+            # NOTE(jamespage)
+            # Accomodate changes in behaviour for pep-0467, introduced
+            # in python 2.7.9.
+            # https://github.com/python/peps/blob/master/pep-0476.txt
+            excp_lst = (TypeError, IOError, ssl.SSLError)
         try:
             HTTPSConnection.__init__(self, host, port,
                                      key_file=key_file,
--- a/tests/test_ssl.py
+++ b/tests/test_ssl.py
@@ -113,8 +113,10 @@ class TestVerifiedHTTPSConnection(testto
         """
         cert_file = os.path.join(TEST_VAR_DIR, 'certificate.crt')
         cacert = os.path.join(TEST_VAR_DIR, 'ca.crt')
+        key_file = os.path.join(TEST_VAR_DIR, 'badkey.key')
         try:
             https.VerifiedHTTPSConnection('127.0.0.1', 0,
+                                          key_file=key_file,
                                           cert_file=cert_file,
                                           cacert=cacert)
             self.fail('Failed to raise assertion.')
